bc_obps/registration/tests/endpoints/v1/_facilities/test_facility_id.py

```diff
@@ -42,30 +107,164 @@ def test_get_facility_with_invalid_facility_id(self):
+        # Create an operator and authorize the current user as an operator user
+        operator, owning_operation = self.create_and_authorize_operator()
+
+        # Create a facility and link it to the owning operation
+        facility = self.create_facility_and_ownership(owning_operation)
 
+        # Act
+        # Make a GET request to the "get_facility" endpoint as an "industry_user"
         response = TestUtils.mock_get_with_auth_role(
             self,
             endpoint=custom_reverse_lazy("get_facility", kwargs={"facility_id": facility.id}),
             role_name="industry_user",
         )
+
+        # Assert
+        # Check that the response status code is 200 (OK)
+        assert response.status_code == 200
+
+        # Check that the facility name is present in the response JSON
+        assert response.json().get('name') is not None
+
+    # PUT
+    def test_facility_endpoint_authorized_users_can_update(self):
+        # Arrange
+        # Create an operator and authorize the current user as an operator user
+        operator, owning_operation = self.create_and_authorize_operator()
+
+        # Create a facility and link it to the owning operation
+        facility = self.create_facility_and_ownership(owning_operation)
+
+        # Construct the URL for the update_facility endpoint with the facility ID
+        url = custom_reverse_lazy("update_facility", kwargs={"facility_id": facility.id})
+
+        # Act
+        # Make a PUT request to the "update_facility" endpoint with the mock payload using the utility method
+        response = TestUtils.mock_put_with_auth_role(
+            self,
+            'industry_user',
+            self.content_type,
+            self.mock_payload | {"operation_id": str(owning_operation.id)},  # Add operation_id to payload
+            url,
+        )
+
+        # Assert
+        # Check that the response status code is 200 (OK)
+        assert response.status_code == 200
+
+        # Check that the facility name in the response JSON matches the updated name
+
+        response_data = response.json()
+
+        # Assert each field to ensure they match the updated values
+        assert response_data.get('name') == "Updated Facility Name"
+        assert response_data.get('type') == "Medium Facility"
+        assert set(response_data.get('well_authorization_numbers')) == {1, 2, 3}
+        assert response_data.get('street_address') == "123 Updated St"
+        assert response_data.get('municipality') == "Updated Municipality"
+        assert response_data.get('province') == "ON"
+        assert response_data.get('postal_code') == "A1A1A1"
+        # Use math.isclose for floating point comparison
+        assert math.isclose(float(response_data.get('latitude_of_largest_emissions')), 43.57, rel_tol=1e-5)
+        assert math.isclose(float(response_data.get('longitude_of_largest_emissions')), -123.58, rel_tol=1e-5)
+
+    def test_facility_endpoint_unauthorized_roles_cannot_update(self):
```

Comment:
Apologies, I am unsure of  the meaning of "checking the count..." Could you expand on your thought?

Reply:
Sure, say we have a facility with an address and 5 well auth numbers, and we want to test that we can change both. I might do something like this after hitting the endpoint:

```
Address.objects.count() == 1 # to make sure we successfully updated the address instead of creating a new one
WellAuthorizationNumber.objects.count() == 2 # to make sure we successfully archived the old ones
```
We could do this in the service instead of the endpoint too